libbeat/statestore/backend/es/registry.go

```diff
@@ -0,0 +1,55 @@
+// Licensed to Elasticsearch B.V. under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Elasticsearch B.V. licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package es
+
+import (
+	"context"
+	"sync"
+
+	"github.com/elastic/beats/v7/libbeat/statestore/backend"
+	"github.com/elastic/elastic-agent-libs/logp"
+)
+
+type Registry struct {
+	ctx context.Context
+
+	log *logp.Logger
+	mx  sync.Mutex
+
+	notifier *Notifier
+}
+
+func New(ctx context.Context, log *logp.Logger, notifier *Notifier) (*Registry, error) {
+	r := &Registry{
+		ctx:      ctx,
+		log:      log,
+		notifier: notifier,
+	}
+
+	return r, nil
+}
```

Comment:
`error` return can be removed

Reply:
Will leave for now as is, as it's consistent with memlog.New that returns the registry and in case if further changes that could return error from the constructor